install_provider.sh

```diff
@@ -0,0 +1,294 @@
+            proxy_http_version                 1.1;
+            proxy_cache_bypass                 \$http_upgrade;
+
+            # Proxy headers
+            proxy_set_header Upgrade           \$http_upgrade;
+            proxy_set_header Connection        \$connection_upgrade;
+            proxy_set_header Host              \$host;
+            proxy_set_header X-Real-IP         \$remote_addr;
+            proxy_set_header X-Forwarded-For   \$proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto \$scheme;
+            proxy_set_header X-Forwarded-Host  \$host;
+            proxy_set_header X-Forwarded-Port  \$server_port;
+
+            # Proxy timeouts
+            proxy_connect_timeout              60s;
+            proxy_send_timeout                 60s;
+            proxy_read_timeout                 60s;
+        }
+
+        # gzip
+        gzip            on;
+        gzip_vary       on;
+        gzip_proxied    any;
+        gzip_comp_level 6;
+        gzip_types      text/plain text/css text/xml application/json application/javascript application/rss+xml application/atom+xml image/svg+xml;
+    }
+}
+EOF
+
+nginx -t || die "nginx doesn't like it's config"
+systemctl restart nginx
+systemctl enable --now reload-nginx.path
+
+# ========= yagna =========
+
+# install erigon runtime before yagna itself
+mkdir -p "${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon"
+cat >"${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon.json" <<EOF
+[
+  {
+    "name": "erigon",
+    "version": "0.1.0",
+    "supervisor-path": "exe-unit",
+    "runtime-path": "ya-runtime-erigon/ya-erigon-runtime",
+    "description": "Service wrapper for Erigon (formelly Turbo-Geth)",
+    "extra-args": ["--runtime-managed-image"]
+  }
+]
+EOF
+curl -sSfL https://github.com/golemfactory/yagna-service-erigon/releases/download/3deaadc/ya-erigon-runtime.tar.gz \
+    | tar -xzf - -C "${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon" \
+    || die "failed to install erigon runtime"
+
+mkdir -p "${ERIGON_DATADIR}/goerli"
+mkdir -p "${ERIGON_USER_HOME}/.local/share/ya-erigon-runtime"
+
+cat >"${ERIGON_USER_HOME}/.local/share/ya-erigon-runtime/ya-erigon-runtime.json" <<EOF
+{
+  "public_addr": "https://${ERIGON_HOSTNAME}:8545",
+  "data_dir": "${ERIGON_DATADIR}",
+  "passwd_tool_path": "htpasswd",
+  "passwd_file_path": "/etc/nginx/erigon_htpasswd",
+  "password_default_length": 15
+}
+EOF
+
+chown -R "${ERIGON_USER}:${ERIGON_USER}" \
+    "${ERIGON_DATADIR}" \
+    "${ERIGON_USER_HOME}/.local/lib/yagna" \
+    "${ERIGON_USER_HOME}/.local/share/ya-erigon-runtime"
+
+# install yagna
+curl -sSf https://join.golem.network/as-provider \
+    | sudo -i -u "$ERIGON_USER" -- env YA_INSTALLER_CORE=pre-rel-v0.7.0-rc6 bash -
+# for some reason even on success installer exits with code 1
+
+sudo -i -u golem ya-provider preset deactivate wasmtime
+sudo -i -u golem ya-provider preset deactivate vm
```

Comment:
Hmm 🤔 - but how we activate the `erigon` runtime ❓ 
It went missed or it's kind of golemsp black magic?

Reply:
It's black magic, although I'm not sure if it's in installer, golemsp, or ya-provider itself. On the first run all runtimes are enabled and installer sets presets for all of them.